src/transformers/models/graphormer/collating_graphormer.py

```diff
@@ -0,0 +1,140 @@
+# Copyright (c) Microsoft Corporation and HuggingFace
+# Licensed under the MIT License.
+
+from typing import Any, Dict, List, Mapping
+
+import numpy as np
+import torch
+
+import pyximport
+
+pyximport.install(setup_args={"include_dirs": np.get_include()})
+from . import algos_graphormer
+
+
+def convert_to_single_emb(x, offset: int = 512):
+    feature_num = x.shape[1] if len(x.shape) > 1 else 1
+    feature_offset = 1 + np.arange(0, feature_num * offset, offset, dtype=np.int64)
+    x = x + feature_offset
+    return x
+
+
+def preprocess_item(item, keep_features=True, task_list=None):
+    task_list = ["y"] if task_list is None else task_list
+
+    # ! If you are using Random Gaussian Features, use keep_features = True
+    if keep_features and "edge_attr" in item.keys():  # edge_attr
+        edge_attr = np.asarray(item["edge_attr"], dtype=np.int64)
+    else:
+        edge_attr = np.ones((len(item["edge_index"][0]), 1), dtype=np.int64)  # same embedding for all
+
+    if keep_features and "node_feat" in item.keys():  # x
+        node_feature = np.asarray(item["node_feat"], dtype=np.int64)
+    else:
+        node_feature = np.ones((item["num_nodes"], 1), dtype=np.int64)  # same embedding for all
+
+    edge_index = np.asarray(item["edge_index"], dtype=np.int64)
+
+    x = convert_to_single_emb(node_feature) + 1
+    N = item["num_nodes"]
```

Comment:
Let's name it `num_nodes`. One letter variables are bad for the reader and capitals are for classes :-)